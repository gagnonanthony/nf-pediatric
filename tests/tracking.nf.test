nextflow_pipeline {

    options "-stub-run"

    name "Test nf-pediatric profile tracking"
    script "../main.nf"

    test("Tracking profile - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking.csv"
                params.outdir = "test_results/"

                params.tracking = true

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 25
        }

    }

    test("Tracking profile no dwi preprocessing - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking.csv"
                params.outdir = "test_results/"

                params.tracking = true
                params.skip_dwi_preprocessing = true

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 17
        }

    }

    test("Tracking profile no rev b0 - should fail") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking_norev.csv"
                params.outdir = "test_results/"

                params.tracking = true

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("Please provide a reverse phase encoded B0 image for sample: sub-test")
        }

    }

    test("Tracking profile no dwi - should fail") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking_nodwi.csv"
                params.outdir = "test_results/"

                params.tracking = true

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("Please provide a DWI image for sample: sub-test")
        }

    }

    test("Tracking profile with infant data - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking_infant.csv"
                params.outdir = "test_results/"

                params.tracking = true
                params.infant = true

                // ** Infant specific parameters ** //
                params.dwi_run_synthstrip               = true
                params.dwi_normalize_fa_mask_threshold  = 0.10
                params.fodf_max_fa_threshold            = 0.01
                params.fodf_min_md_threshold            = 0.00185
                params.frf_manual_frf                   = "12,5,5"
                params.run_pft_tracking                 = false
                params.local_min_len                    = 15
                params.local_fa_seeding_mask_threshold  = 0.10
                params.run_local_tracking               = true

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 28
        }

    }

    test("Tracking profile with infant data no wm mask - should fail") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking_infant_nowmmask.csv"
                params.outdir = "test_results/"

                params.tracking = true
                params.infant = true

                // ** Infant specific parameters ** //
                params.dwi_run_synthstrip               = true
                params.dwi_normalize_fa_mask_threshold  = 0.10
                params.fodf_max_fa_threshold            = 0.01
                params.fodf_min_md_threshold            = 0.00185
                params.frf_manual_frf                   = "12,5,5"
                params.run_pft_tracking                 = false
                params.local_min_len                    = 15
                params.local_fa_seeding_mask_threshold  = 0.10

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("Please provide a wmparc image for sample: sub-test")
        }

    }

    test("Tracking profile with infant data no rev b0 - should fail") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking_infant_norev.csv"
                params.outdir = "test_results/"

                params.tracking = true
                params.infant = true

                // ** Infant specific parameters ** //
                params.dwi_run_synthstrip               = true
                params.dwi_normalize_fa_mask_threshold  = 0.10
                params.fodf_max_fa_threshold            = 0.01
                params.fodf_min_md_threshold            = 0.00185
                params.frf_manual_frf                   = "12,5,5"
                params.run_pft_tracking                 = false
                params.local_min_len                    = 15
                params.local_fa_seeding_mask_threshold  = 0.10

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("Please provide a reverse phase encoded B0 image for sample: sub-test")
        }

    }

    test("Tracking profile with infant data no t2 - should fail") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking_infant_not2.csv"
                params.outdir = "test_results/"

                params.tracking = true
                params.infant = true

                // ** Infant specific parameters ** //
                params.dwi_run_synthstrip               = true
                params.dwi_normalize_fa_mask_threshold  = 0.10
                params.fodf_max_fa_threshold            = 0.01
                params.fodf_min_md_threshold            = 0.00185
                params.frf_manual_frf                   = "12,5,5"
                params.run_pft_tracking                 = false
                params.local_min_len                    = 15
                params.local_fa_seeding_mask_threshold  = 0.10

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("Please provide at least one anatomical image (T1w or T2w) for sample: sub-test")
        }

    }

}
