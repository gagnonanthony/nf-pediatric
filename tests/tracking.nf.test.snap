{
    "Tracking profile - should run successfully": {
        "content": [
            30,
            {
                "DENOISE_DWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "REGISTRATION_ANATTODWI": {
                    "ants": "2.4.3"
                },
                "SEGMENTATION_FASTSEG": {
                    "scilpy": "2.0.0",
                    "fsl": 6.0
                },
                "TRACKING_LOCALTRACKING": {
                    "scilpy": "2.0.2"
                },
                "TRACKING_PFTTRACKING": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_desc-boundingbox.pkl",
                "sub-test/anat/sub-test_desc-brain_mask.nii.gz",
                "sub-test/anat/sub-test_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_from-T1w_to-dwi_affine.mat",
                "sub-test/anat/sub-test_from-T1w_to-dwi_warp.nii.gz",
                "sub-test/anat/sub-test_from-dwi_to-T1w_warp.nii.gz",
                "sub-test/anat/sub-test_space-DWI_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_probseg.nii.gz",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-ad.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_max.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_sum.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_total.nii.gz",
                "sub-test/dwi/sub-test_desc-brain_mask.nii.gz",
                "sub-test/dwi/sub-test_desc-csf_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-fa.nii.gz",
                "sub-test/dwi/sub-test_desc-fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-ga.nii.gz",
                "sub-test/dwi/sub-test_desc-gm_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-local_tracking.trk",
                "sub-test/dwi/sub-test_desc-md.nii.gz",
                "sub-test/dwi/sub-test_desc-mode.nii.gz",
                "sub-test/dwi/sub-test_desc-nufo.nii.gz",
                "sub-test/dwi/sub-test_desc-peaks.nii.gz",
                "sub-test/dwi/sub-test_desc-pft_tracking.trk",
                "sub-test/dwi/sub-test_desc-preproc_b0.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bval",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bvec",
                "sub-test/dwi/sub-test_desc-preproc_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-rd.nii.gz",
                "sub-test/dwi/sub-test_desc-rgb.nii.gz",
                "sub-test/dwi/sub-test_desc-tensor.nii.gz",
                "sub-test/dwi/sub-test_desc-wm_fodf.nii.gz",
                "sub-test/dwi/sub-test_frf.txt",
                "sub-test/dwi/sub-test_space-DWI_label-exclude_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-include_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-local_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-pft_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-tracking_desc-local_mask.nii.gz",
                "sub-test/figures",
                "sub-test/figures/sub-test__coverage_overlay_mqc.png",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_metrics_mqc.png",
                "sub-test/figures/sub-test_tissue_segmentation_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-20T10:28:31.622283"
    },
    "Tracking profile with infant data only T2 - should run successfully": {
        "content": [
            26,
            {
                "DENOISE_DWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-ad.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_max.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_sum.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_total.nii.gz",
                "sub-test/dwi/sub-test_desc-brain_mask.nii.gz",
                "sub-test/dwi/sub-test_desc-csf_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-fa.nii.gz",
                "sub-test/dwi/sub-test_desc-fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-ga.nii.gz",
                "sub-test/dwi/sub-test_desc-gm_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-md.nii.gz",
                "sub-test/dwi/sub-test_desc-mode.nii.gz",
                "sub-test/dwi/sub-test_desc-nufo.nii.gz",
                "sub-test/dwi/sub-test_desc-peaks.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_b0.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bval",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bvec",
                "sub-test/dwi/sub-test_desc-preproc_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-pwdavg_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-rd.nii.gz",
                "sub-test/dwi/sub-test_desc-rgb.nii.gz",
                "sub-test/dwi/sub-test_desc-tensor.nii.gz",
                "sub-test/dwi/sub-test_desc-wm_fodf.nii.gz",
                "sub-test/dwi/sub-test_frf.txt",
                "sub-test/figures",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_metrics_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-20T10:32:12.735499"
    },
    "Tracking profile with infant data T2 and T1 - should run successfully": {
        "content": [
            26,
            {
                "DENOISE_DWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-ad.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_max.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_sum.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_total.nii.gz",
                "sub-test/dwi/sub-test_desc-brain_mask.nii.gz",
                "sub-test/dwi/sub-test_desc-csf_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-fa.nii.gz",
                "sub-test/dwi/sub-test_desc-fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-ga.nii.gz",
                "sub-test/dwi/sub-test_desc-gm_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-md.nii.gz",
                "sub-test/dwi/sub-test_desc-mode.nii.gz",
                "sub-test/dwi/sub-test_desc-nufo.nii.gz",
                "sub-test/dwi/sub-test_desc-peaks.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_b0.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bval",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bvec",
                "sub-test/dwi/sub-test_desc-preproc_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-pwdavg_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-rd.nii.gz",
                "sub-test/dwi/sub-test_desc-rgb.nii.gz",
                "sub-test/dwi/sub-test_desc-tensor.nii.gz",
                "sub-test/dwi/sub-test_desc-wm_fodf.nii.gz",
                "sub-test/dwi/sub-test_frf.txt",
                "sub-test/figures",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_metrics_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-20T10:33:50.362463"
    },
    "Tracking profile no dwi preprocessing - should run successfully": {
        "content": [
            22,
            {
                "CROPDWI": {
                    "scilpy": "2.0.2"
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "EXTRACTB0": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "REGISTRATION_ANATTODWI": {
                    "ants": "2.4.3"
                },
                "RESAMPLE_DWI": {
                    "scilpy": "2.0.2"
                },
                "SEGMENTATION_FASTSEG": {
                    "scilpy": "2.0.0",
                    "fsl": 6.0
                },
                "TRACKING_LOCALTRACKING": {
                    "scilpy": "2.0.2"
                },
                "TRACKING_PFTTRACKING": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_desc-boundingbox.pkl",
                "sub-test/anat/sub-test_desc-brain_mask.nii.gz",
                "sub-test/anat/sub-test_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_from-T1w_to-dwi_affine.mat",
                "sub-test/anat/sub-test_from-T1w_to-dwi_warp.nii.gz",
                "sub-test/anat/sub-test_from-dwi_to-T1w_warp.nii.gz",
                "sub-test/anat/sub-test_space-DWI_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_probseg.nii.gz",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-ad.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_max.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_sum.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_total.nii.gz",
                "sub-test/dwi/sub-test_desc-brain_mask.nii.gz",
                "sub-test/dwi/sub-test_desc-csf_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-fa.nii.gz",
                "sub-test/dwi/sub-test_desc-fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-ga.nii.gz",
                "sub-test/dwi/sub-test_desc-gm_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-local_tracking.trk",
                "sub-test/dwi/sub-test_desc-md.nii.gz",
                "sub-test/dwi/sub-test_desc-mode.nii.gz",
                "sub-test/dwi/sub-test_desc-nufo.nii.gz",
                "sub-test/dwi/sub-test_desc-peaks.nii.gz",
                "sub-test/dwi/sub-test_desc-pft_tracking.trk",
                "sub-test/dwi/sub-test_desc-preproc_b0.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bval",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bvec",
                "sub-test/dwi/sub-test_desc-preproc_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-rd.nii.gz",
                "sub-test/dwi/sub-test_desc-rgb.nii.gz",
                "sub-test/dwi/sub-test_desc-tensor.nii.gz",
                "sub-test/dwi/sub-test_desc-wm_fodf.nii.gz",
                "sub-test/dwi/sub-test_frf.txt",
                "sub-test/dwi/sub-test_space-DWI_label-exclude_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-include_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-local_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-pft_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-tracking_desc-local_mask.nii.gz",
                "sub-test/figures",
                "sub-test/figures/sub-test__coverage_overlay_mqc.png",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_metrics_mqc.png",
                "sub-test/figures/sub-test_tissue_segmentation_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-20T10:29:50.377023"
    }
}